concordium-smart-contract-testing/src/types.rs

```diff
@@ -600,3 +641,119 @@ pub struct ExchangeRateError;
 #[derive(Debug, Error)]
 #[error("Any signer must have at least one key.")]
 pub struct ZeroKeysError;
+
+/// Errors that occur while setting up the connection to an external node.
+#[derive(Debug, thiserror::Error)]
+pub enum SetupExternalNodeError {
+    /// It was not possible to connect to a node on the provided endpoint.
+    #[error("Could not connect to the provided endpoint due to: {error}")]
+    CannotConnect {
+        /// The inner error.
+        #[from]
+        error: sdk::endpoints::Error,
+    },
+    /// The attempt to connect to an external node timed out.
+    #[error("The attempt to connect to an external node timed out.")]
+    ConnectTimeout,
+    /// The query to check the `external_query_block` timed out.
+    #[error("The query to check the `external_query_block` timed out.")]
+    CheckQueryBlockTimeout,
+    /// The specified external query block does not exist.
+    #[error("The specified external query block {query_block} does not exist.")]
+    QueryBlockDoesNotExist { query_block: BlockHash },
+    /// Could not check the existence of the specified query block or the last
+    /// final block.
+    #[error(
+        "Could not check the existence of the specified query block or the last final block due \
+         to: {error}"
+    )]
+    CannotCheckQueryBlockExistence {
+        /// The inner error.
+        error: sdk::v2::RPCError,
+    },
+}
+
+/// Errors that occur while trying to communicate with an external node.
+#[derive(Debug, Error)]
+pub enum ExternalNodeError {
+    /// An external node has not been configured.
+    #[error("An external node has not been configured.")]
+    NotConfigured,
+    /// The query could not be performed.
+    #[error("Could not perform the query: {error}")]
+    QueryError {
+        #[from]
+        error: sdk::endpoints::QueryError,
+    },
+    /// The query timed out.
+    #[error("The query timed out.")]
+    QueryTimeout,
+}
+
+/// The error returned when an external node has not been configured prior to
+/// using it.
+#[derive(Debug, Error, PartialEq, Eq)]
+#[error("An external node has not been configured.")]
+pub struct ExternalNodeNotConfigured;
+
+#[derive(Debug, Error)]
+pub enum ChainBuilderError {
+    /// The provided exchange rates are not valid.
+    /// Meaning that they do not correspond to one energy costing less than
+    /// `u64::MAX / `
+    /// [`concordium_base::constants::MAX_ALLOWED_INVOKE_ENERGY`]`.
```

Comment:
This constant is no longer in base

Reply:
Hmm, you are right. I see that Ales removed it again. Has it been moved elsewhere, or why was it removed @abizjak ? 😊

Reply:
It was removed because the node API has no such concept anymore and it does not even require you to supply `max_energy`. 

Instead the node itself has a configuration option to prevent DOS and it will decide the maximum energy allowed.

YOu can still supply an NRG amount if you want to test something, but the NRG amount that will actually be used is min(supplied amount, node configuration option).

Reply:
As discussed with Ales, I've replaced the constant with the value instead,